### ISSUE TITLE
Media: Reports of images being incorrectly resized

### ISSUE THREAD

**Issue by aerych** [wordpress-mobile/WordPress-iOS#5332]
As [reported in the forums](https://ios.forums.wordpress.org/topic/max-image-upload-size-incorrect):

> I'm also noticing that the "max image upload size" setting no longer works properly since the last update.
> If I set it to anything other than "original", the uploaded size is always 374x280. The originals are taken on an iPhone 6S and are, of course, much larger than that.  
> 
> I have the max size settings on 800x800 and images end up 374x280. Originals are much bigger. No media/gallery plugin installed. With Iphone 5s it is working. Images that end up 374x280 are rezised in in Iphone native Photo app. Not resized once end up right. Do not know why. Media setting in the blog are at max 1024x.

**Comment by SergioEstevao** [comment]
@aerych where you able to reproduce this issue? I've tried with different sizes but that never happens to me.

**Comment by aerych** [comment]
I was not able to reproduce the issue.  Since we have multiple reports in the forums I'm assuming there is some hidden x-factor to trigger the glitch.

**Comment by rachelsquirrel** [comment]
We've also had a report of strange image sizing in Helpshift which might be related #98335.

**Comment by kurzee** [comment]
Looks this was resolved in recent work for `release/6.7`, originally this was a self-hosted only issue.

**Comment by aerych** [comment]
Reopening this one.  A [recent forum post](https://ios.forums.wordpress.org/topic/photo-upload-size-374x280px?replies=4) indicates the bug is still lurking tho I am still unable to reproduce it when It try. 

From the forum thread:

> When uploading several photos from the Wordpress mobile app (v6.7 on iOS 10.2) to my self hosted blog, only the first or some of the images will respect my "Media Max Image Upload Size" settings of 1200x1200px but additional photos are uploaded in the very tiny size of 374x280px. I can't figure out why the selected batch of photos won't upload all in the same 1200x1200px size. Please Help! Thanks

and

> The uploaded images show up in the wp-admin Media library with the same small dimensions of 374x280px. see screenshot. Uploading several images with the Wordpress app seems to me pretty quick. It looks like it is uploading only the first (and sometimes the second image) in the desired image size, all the following other images are scaled down to the mentioned mini size of 374x280px explaining the fast process of uploading. Only uploading one by one fixes the image size problem. Hope that helps to find the bug?
> 
> [screenshot](https://www.dropbox.com/s/bieunllbmt2y2up/screenshot.png?dl=0)


Note: It looks like the forum thread in the original report is mixed up with some other content after the wporg forum migrations. :/

**Comment by aerych** [comment]
@kurzee was there something specific in release/6.7 that seemed to address the original report?

**Comment by koke** [comment]
Multiple reports show that size of `374x280`, which I found rather curious. I haven't managed to reproduce it, but my current hypothesis is that we are uploading the image thumbnail, which has a size that matches the screen (375 points wide on iPhone 6).

@SergioEstevao any ideas on how to reproduce that? I've seen [some code](https://github.com/wordpress-mobile/WordPress-iOS/blob/0d18ec5366722b1eb94f72c6f0c755c2f4b146c4/WordPress/Classes/Services/MediaService.m#L517) that tries to use the thumbnail if it's large enough, but I haven't been able to trigger it.

**Comment by daniloercoli** [comment]
Another user reported the same issue.

> Some images uploaded fine, in their original resolution. Others were resized down to 374x280px. I was able to re-upload some of them later at the correct resolution. But one or two seem stubbornly stuck at the small res, even after multiple attempts. I confirmed that the originals are indeed 2048×1536, so I wasn’t mistakenly uploading small images.

**Comment by kurzee** [comment]
I've tested this a bit more and also cannot get this to reproduce with `6.9`. I'm not too familiar with the media code here but we'll certainly put this on our radar with the upcoming media work with @SergioEstevao.

>@kurzee was there something specific in release/6.7 that seemed to address the original report?

IIRC, there wasn't a particular line of code that I had eyeballed, but that several blocks of media code had been refactored or converted. But I do remember testing the original issue and reproducing it on the `release/6.6` branch and not on the `release/6.7` which had the refactoring work, at the time. I should have looked into it a bit more and documented ☹️
